oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsUsbDevice.java

```diff
@@ -55,269 +57,256 @@
  */
 public class WindowsUsbDevice extends AbstractUsbDevice {
 
-    private static final Logger LOG = LoggerFactory.getLogger(WindowsUsbDevice.class);
+	private static final Logger LOG = LoggerFactory.getLogger(WindowsUsbDevice.class);
 
-    enum USBControllerProperty {
-        PNPDEVICEID;
-    }
+	enum USBControllerProperty {
+		PNPDEVICEID;
+	}
 
-    // List of controllers
-    private static List<String> controllerDeviceIds = null;
+	// List of controllers
+	private static Supplier<List<String>> controllerDeviceIds = Memoizer.memoize(WindowsUsbDevice::getControllerDeviceIdList);
 
-    enum PnPEntityProperty {
-        NAME, MANUFACTURER, PNPDEVICEID;
-    }
+	enum PnPEntityProperty {
+		NAME, MANUFACTURER, PNPDEVICEID;
+	}
 
-    private static final String PNPENTITY_BASE_CLASS = "Win32_PnPEntity";
+	private static final String PNPENTITY_BASE_CLASS = "Win32_PnPEntity";
 
-    enum DiskDriveProperty {
-        PNPDEVICEID, SERIALNUMBER;
-    }
+	enum DiskDriveProperty {
+		PNPDEVICEID, SERIALNUMBER;
+	}
 
-    private static final String DISKDRIVE_BASE_CLASS = "Win32_DiskDrive";
+	private static final String DISKDRIVE_BASE_CLASS = "Win32_DiskDrive";
 
-    private static final Pattern VENDOR_PRODUCT_ID = Pattern
-            .compile(".*(?:VID|VEN)_(\\p{XDigit}{4})&(?:PID|DEV)_(\\p{XDigit}{4}).*");
+	private static final Pattern VENDOR_PRODUCT_ID = Pattern
+			.compile(".*(?:VID|VEN)_(\\p{XDigit}{4})&(?:PID|DEV)_(\\p{XDigit}{4}).*");
 
-    /**
-     * <p>
-     * Constructor for WindowsUsbDevice.
-     * </p>
-     *
-     * @param name
-     *            a {@link java.lang.String} object.
-     * @param vendor
-     *            a {@link java.lang.String} object.
-     * @param vendorId
-     *            a {@link java.lang.String} object.
-     * @param productId
-     *            a {@link java.lang.String} object.
-     * @param serialNumber
-     *            a {@link java.lang.String} object.
-     * @param uniqueDeviceId
-     *            a {@link java.lang.String} object.
-     * @param connectedDevices
-     *            an array of {@link oshi.hardware.UsbDevice} objects.
-     */
-    public WindowsUsbDevice(String name, String vendor, String vendorId, String productId, String serialNumber,
-            String uniqueDeviceId, UsbDevice[] connectedDevices) {
-        super(name, vendor, vendorId, productId, serialNumber, uniqueDeviceId, connectedDevices);
-    }
+	/**
+	 * <p>
+	 * Constructor for WindowsUsbDevice.
+	 * </p>
+	 *
+	 * @param name             a {@link java.lang.String} object.
```

Comment:
This is another formatting convention that OSHI uses with the newline/indent for parameters.